src/kOS/Execution/CPU.cs

```diff
@@ -1031,7 +1037,12 @@ private void ProcessTriggers()
                 }
             }
 
-            currentContext.InstructionPointer = currentInstructionPointer;
+            // since `run` opcodes don't work in triggers, we can use the opcode count to determine if the
+            // program has been aborted.  If the count isn't right, leave the pointer where it is.
```

Comment:
I'm not comfortable with layering a good feature on top of a bad misfeature.  It's true we can't RUN from a trigger, but that's unfortunate, not desirable.  Making use of that as a core assumption in implementing something else useful makes it harder to fix the misfeature later.